os/arch/arm/src/amebasmart/amebasmart_smp.c

```diff
@@ -73,6 +74,109 @@ void rtk_core1_power_off(void)
 	HAL_WRITE32(SYSTEM_CTRL_BASE_HP, REG_HSYS_HP_PWC, val);
 }
 
+void vPortSecondaryStart(int cpu)
+{
+        int count = 10;
+        int state;
+
+        if(cpu != 1) {
+                lldbg("cpu number out of range");
+                return;
+        }
+
+#ifdef CONFIG_CPU_HOTPLUG
+        /* take the lock, as we are about to put cpu1 idle task back active */
+        sched_lock();
+
+        /* turn on CPU1's power domain, require time to stabilize */
+        rtk_core1_power_on();
+        DelayMs(1);
+
+        /* renable scu and invalidate tags on all processors */
+        arm_enable_smp(0);
+
+        cpu_set_state(cpu, CPU_HOTPLUG);
+
+        /* cold-boot the core */
+        lldbg("Booting secondary core... \n");
+        BaseType_t err = psci_cpu_on(cpu, (unsigned long)__cpu1_start);
+        DEBUGASSERT(err >= 0);
+
+        /* await for PSCI to report cpu state */
+        do {
+                state  = psci_affinity_info(cpu, 0);
+                if (state == 1) {
+                        break;
+                }
+
+                DelayUs(50);
+        } while (count--);
+
+        lldbg("Secondary core booted?... %d\n", cpu_get_state(cpu));
+        cpu_set_state(cpu, CPU_SCHED_OFF);
+
+        /* fire SGI1 to kick the freshly booted CPU into TizenRT Idle Task */
+        up_cpu_start(cpu);
+
+        /* cpu1 is now alive again and can be scheduled on */
+        sched_unlock();
+#endif
+}
+
+void smp_full_powerdown_cpu(int cpu)
+{
+	int count = 10; // 10 * 50us = 500us
+	int state;
+
+	if(cpu != 1) {
+		lldbg("cpu number out of range");
+		return;
+	}
+
+#ifdef CONFIG_CPU_HOTPLUG
+	smp_hotplug_cpu(cpu);
+
+	lldbg("Secondary core shutdown?... %d\n", cpu_get_state(cpu));
+
+	/* wait for PSCI to report that CPU1 is now safe to power off */
+	do {
+		state  = psci_affinity_info(cpu, 0);
+		if (state == 1) {
+			cpu_set_state(cpu, CPU_HALTED);
+
+			lldbg("Secondary core shutdown complete: %d\n", cpu_get_state(cpu));
+			/* turn off CPU1's power domain */
+			rtk_core1_power_off();
+			return;
+		}
+
+		DelayUs(50);
+	} while (count--);
+#endif
+}
```

Comment:
For CPU On/Off operations, it is more appropriate for the OS to utilize the necessary BSP functionalities rather than the BSP using OS functions.

If we continue with the current code, whenever we port a new chipset, we will need to request the vendor to provide information and requirements regarding the Tizen Lite CPU On/Off flow.

I suggest split this function up_cpu_hotplug and BSP side core power down(PSCI off and power block off)

Reply:
yes, i think, current we don't PSCI layer so,  `2.core power down (rtk power off)` has PSCI off also is better